app/assets/javascripts/discourse/app/lib/uppy-checksum-plugin.js

```diff
@@ -57,38 +59,41 @@ export default class UppyChecksum extends Plugin {
               .map((b) => b.toString(16).padStart(2, "0"))
               .join("");
             this.uppy.setFileMeta(fileId, { sha1_checksum: hashHex });
+            this.uppy.emit("preprocess-complete", this.pluginClass, file);
```

Comment:
is uppy supposed to be public?

Reply:
Yes this is how Uppy stores a reference to the uppy instance when initializing a plugin